### DISCUSSION TITLE
Add number of rows affected to data properties of DB::listen()

### DISCUSSION THREAD

**Discussion by isaiahfisher** [laravel/framework#52457] [Ideas]
Recently, I am working to find a query that is affecting more rows than intended. My plan is to use the DB::listen() function to trace and log queries in the application. While I should be able to find the query without the number of rows affected being listed, I believe this is a valuable metric to know when logging queries.

My proposal is to add a data property to retrieve the count of rows that were affected by the query in this function. This count helps provide a clearer picture of the impact of the query on the underlying data.

**Comment by shaedrich** [comment]
As far as I know, `DB::listen()` is quite limited. Maybe, additionally, there should be a second parameter with access to the output rather than the input. The `time` property should go to the output then. Or `QueryExecuted` is restructured to have an `input` and an `output` property with the actual properties thereunder.
